### PR TITLE
Detect `context.projectMetadata.hasRouter` and send to the index

### PR DESCRIPTION
**Pull request by tmeasday**
This PR is blocked on an index change (to come).

To QA:

- Create a routing based project (e.g. SB Next Sandbox)
- Run a build, check it sets `hasRouter=true`
- Create a non-routed project (e.g. `react-vite` Sandbox)
- Run a build, check it sets `hasRouter=false`

Use this review app: https://github.com/chromaui/chromatic/pull/9061

<details>
  <summary>📦 Published PR as canary version: <code>11.17.0--canary.1112.11702183022.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install chromatic@11.17.0--canary.1112.11702183022.0
  # or 
  yarn add chromatic@11.17.0--canary.1112.11702183022.0
  ```
</details>

### CODE REVIEW

**Comment by tmeasday** [comment]
I plan to add some other fields to `projectMetadata`. Would you prefer one PR or many @codykaup ?

**Comment by codykaup** [comment]
> I plan to add some other fields to `projectMetadata`. Would you prefer one PR or many @codykaup ?

Totally your call @tmeasday ! I usually base it on the size and concept of the change. I think since you're adding a bunch of metadata _and_ there's a timezone difference, I'm cool with a single PR for it!

File: node-src/tasks/gitInfo.ts [hunk 1]
```diff
@@ -260,6 +269,13 @@ export const setGitInfo = async (ctx: Context, task: Task) => {
     }
   }
 
+  ctx.projectMetadata = {
+    hasRouter: getHasRouter(ctx.packageJson),
+    creationDate: await getRepositoryCreationDate(),
+    numberOfCommitters: await getNumberOfComitters(),
+    numberOfAppFiles: await getCommittedFileCount(['page', 'screen'], ['js', 'jsx', 'ts', 'tsx']),
+  };
```
Comment (codykaup) [inline node-src/tasks/gitInfo.ts:277]: There are cases where TurboSnap returns early. Should we move this up so we aren't missing this metadata when that happens?

**Review by codykaup — COMMENTED** [review]
Code looks good other than the placement of setting `ctx.projectMetadata`!

**Review by codykaup — APPROVED** [review]

**Pull request merged**